### ISSUE TITLE
TypeError: '>' not supported between instances of 'int' and 'list'

### ISSUE THREAD

**Issue by bowlofeggs** [fedora-infra/bodhi#2768]
Another composer crash, possibly related to the Python 3 deployment:

```
[2018-11-21 14:30:50][fedmsg.consumers   ERROR]epel7-testing Exception in ComposerThread(epel7-testing)
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/bodhi/server/consumers/masher.py", line 411, in work
    self._compose_updates()
  File "/usr/lib/python3.7/site-packages/bodhi/server/consumers/masher.py", line 942, in _compose_updates
    self.generate_testing_digest()
  File "/usr/lib/python3.7/site-packages/bodhi/server/consumers/masher.py", line 704, in generate_testing_digest
    self.add_to_digest(update)
  File "/usr/lib/python3.7/site-packages/bodhi/server/consumers/masher.py", line 696, in add_to_digest
    update, use_template='maillist_template')):
  File "/usr/lib/python3.7/site-packages/bodhi/server/mail.py", line 359, in get_template
    (to_unicode(build.get_changelog(oldtime)), line)
  File "/usr/lib/python3.7/site-packages/bodhi/server/models.py", line 1634, in get_changelog
    while (i < num) and (when[i] > timelimit):
TypeError: '>' not supported between instances of 'int' and 'list'
```

**Comment by bowlofeggs** [comment]
Welp, I now have a reproducer:

```
from threading import Lock

import koji

from bodhi.server import buildsys, config, mail, models, initialize_db


initialize_db(config.config)
buildsys._buildsystem = lambda: koji.ClientSession('https://koji.fedoraproject.org/kojihub')
buildsys._buildsystem_login_lock = Lock()

b = models.Build.query.filter_by(nvr='libpeas-loader-python3-1.22.0-1.el7').one()
print(mail.get_template(b.update))
```

**Comment by bowlofeggs** [comment]
The issue seems to be [here](https://github.com/fedora-infra/bodhi/blob/3.11.0/bodhi/server/mail.py#L350-L359):

```
            oldh = get_rpm_header(lastpkg)
            oldtime = oldh['changelogtime']
            text = oldh['changelogtext']
            del oldh
            if not text:
                oldtime = 0
            elif len(text) != 1:
                oldtime = oldtime[0]
            info['changelog'] = u"ChangeLog:\n\n%s%s" % \
                (to_unicode(build.get_changelog(oldtime)), line)
```

With the package given above, there is a ```lastpkg```, and ```oldh``` looks like this:

```
{'description': 'This package contains the Python 3 loader that is needed to\nrun Python 3 plugins that use libpeas.', 'url': 'https://wiki.gnome.org/Projects/Libpeas', 'changelogname': ['Björn Esser <besser82@fedoraproject.org> - 1.20.0-1'], 'summary': 'Python 3 loader for libpeas', 'version': '1.20.0', 'changelogtime': [1510920000], 'changelogtext': ['- Initial rpm release for EPEL7 (rhbz#1514458)'], 'release': '1.el7', 'name': 'libpeas-loader-python3'}
```

Note that ```changelogtime``` is a list of one element, so ```oldtime``` is set to ```[1510920000]```. ```text``` is ```['- Initial rpm release for EPEL7 (rhbz#1514458)']``` and so has one element, so nothing ever converts the single element list to an int before passing it to ```build.get_changelog()```, which results in comparing an int to a list.

**Comment by bowlofeggs** [comment]
The fix for this issue is now released with Bodhi 3.11.1 and deployed to Fedora's production instance.
